app/components/category-item.js

```diff
@@ -15,4 +18,37 @@ export default Ember.Component.extend({
     let userCategories = this.get('userCategories');
     return userCategories.hasCategory(category);
   }),
+
+  actions: {
+    addCategory(category) {
+      this.set('isLoading', true);
+      this.addCategory(category).catch(() => {
```

Comment:
Something like the latter, the `user-category` service, maybe makes the most sense since the data is being handled there anyway, and the way we add/remove a category via the API will not change implementation across UIs (like a user's profile editing, for instance). There's no need to have that happen in the controller.

`{{categories-list categories=model addCategory=(action 'addCategory') removeCategory=(action 'removeCategory')}}` <= and that is weird. It works right now, but doesn't mean it's right. Ostensibly we would pass _nothing_ into the component, but inside the component itself we can handle the `addCategory` action and then call out to the `userCategory` service.